examples/plot_credit_loan_decisions.py

```diff
@@ -0,0 +1,1293 @@
+        - Epsilon: Float representing maximum difference bound for the fairness Moment constraint
+
+    Returns:
+        - Predictors: List of inner model predictors learned by the
+        ExponentiatedGradient model during the training process.
+
+    """
+    exp_grad_est = ExponentiatedGradient(
+        estimator=estimator.named_steps["classifier"],
+        constraints=EqualizedOdds(difference_bound=epsilon),
+    )
+    # Is this an issue - Re-runs
+    exp_grad_est.fit(X_train, y_train, sensitive_features=A_train)
+    predictors = exp_grad_est.predictors_
+    return predictors
+
+
+# %%
+# Because the *performance-fairness trade-off* learned by the
+# :code:`ExponentiatedGradient` model is sensitive to our chosen :code:`epsilon` value, we
+# can treat :code:`epsilon` as a *hyperparameter* and iterate over a range of
+# potential values. Here, we will train two :code:`ExponentiatedGradient` models, one
+# with :code:`epsilon=0.01` and the second with :code:`epsilon=0.02`, and store the inner
+# models learned through each of the training processes.
+#
+# In practice, we recommend choosing smaller values of :code:`epsilon` on
+# the order of the *square root* of the number of samples in the training
+# dataset: :math:`\dfrac{1}{\sqrt{\text{numberSamples}}} \approx \dfrac{1}{\sqrt{25000}}
+# \approx 0.01`
+
+# %%
+epsilons = [0.01, 0.02]
+
+# %%
+all_models = {}
+for eps in epsilons:
+    all_models[eps] = get_expgrad_models_per_epsilon(
+        estimator=estimator,
+        epsilon=eps,
+        X_train=X_train,
+        y_train=y_train,
+        A_train=A_train,
+    )
+
+# %%
+for epsilon, models in all_models.items():
+    print(
+        f"For epsilon {epsilon}, ExponentiatedGradient learned {len(models)} inner models"
+    )
+
+
+# %%
+# Here, we can see all the inner models learned for each value of
+# :code:`epsilon`. With the :code:`ExponentiatedGradient` model, we specify an
+# :code:`epsilon` parameter that represents the maximal disparity in our fairness
+# metric that our final model should satisfy. For example, an :code:`epsilon=0.02`
+# means that the training value of the *equalized odds difference* of the
+# returned model is at most :code:`0.02` (if the algorithm converges).
+
+# %%
+# Reviewing inner models of ExponentiatedGradient
+# ===============================================
+# In many situations due to regulation or other technical restrictions, the
+# randomized nature of :code:`ExponentiatedGradient` algorithm may be undesirable. In
+# addition, the multiple inner models of the algorithm introduce challenges for
+# model interpretability. One potential workaround to avoid these issues is to
+# select one of the inner models and deploy it instead.
+#
+# In the previous section, we trained multiple :code:`ExponentiatedGradient` models
+# at different :code:`epsilon` levels and collected all the inner models learned by
+# this process. However, some of these inner models are better performance in
```

Comment:
This sentence doesn't quite make sense. I changed it from "are better performance" to "have worse performance"
Wdyt @MiroDudik  @LeJit ?

Reply:
i have edited this, so it's now fixed.